flint-core/src/main/scala/org/opensearch/flint/core/storage/FlintOpenSearchIndexMetadataService.scala

```
@@ -191,6 +194,15 @@ object FlintOpenSearchIndexMetadataService {
                   }
                 }
               }
+            case "_source" =>
+              parseObjectField(parser) { (parser, innerFieldName) =>
+                {
+                  innerFieldName match {
+                    case "enabled" => builder.indexMappingsSourceEnabled(parser.booleanValue())
+                    case _ => // Handle other fields as needed
```

Is this TODO comment? should we raise an error in case of unknown value?

---

Yes, currently we ignore other values, will add more values if needed.

---

There's some other values allowed in _source. This PR only handle "enabled": https://docs.opensearch.org/docs/latest/field-types/metadata-fields/source/

But I think there's room for generalizing the change to allow for all these cases for `_source`, without limiting to enabled.